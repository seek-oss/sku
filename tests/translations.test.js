const path = require('path');
const runSkuScriptInDir = require('../test/utils/runSkuScriptInDir');
const { getAppSnapshot } = require('../test/utils/appSnapshot');
const waitForUrls = require('../test/utils/waitForUrls');

const appDir = path.dirname(
  require.resolve('@fixtures/translations/sku.config.js'),
);
const { port } = require('@fixtures/translations/sku.config.js');

const baseUrl = `http://localhost:${port}`;

describe('translations', () => {
  let process;

  beforeAll(async () => {
    await runSkuScriptInDir('build', appDir);
    process = await runSkuScriptInDir('serve', appDir);
    await waitForUrls(`${baseUrl}/en`);
  });

  afterAll(() => {
    process.kill();
  });

  it('should render en', async () => {
    const app = await getAppSnapshot(`${baseUrl}/en`);
    expect(app).toMatchSnapshot();
  });

  it('should render fr', async () => {
    const app = await getAppSnapshot(`${baseUrl}/fr`);
    expect(app).toMatchSnapshot();
  });

  it('should render en-PSEUDO post-hydration', async () => {
    const app = await getAppSnapshot(`${baseUrl}/en?pseudo=true`);
    expect(app).toMatchSnapshot();
  });

  it('should support query parameters', async () => {
    const app = await getAppSnapshot(`${baseUrl}/en?a=1`);
    expect(app).toMatchSnapshot();
  });
});
