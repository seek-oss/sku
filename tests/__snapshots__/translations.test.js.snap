// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`translations should render en 1`] = `
SCRIPTS: [
  "/runtime-c90bedc6c1e41d593d60.js",
  "/298-e7bbd14d05c894fe24a5.js",
  "/main-952ac1a7a163db10b764.js",
  "/en-translations-b17efc1aa5d571ed56f2.js",
]
CSS: []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[0]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[1]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[2]"
    >
    <link
      data-chunk="en-translations"
      rel="preload"
      as="script"
      href="scripts[3]"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Hello sku
        Company name
      </div>
    </div>
    <script
      id="__SKU_CLIENT_CONTEXT__"
      type="application/json"
    >
      {"language":"en"}
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
    >
      [938]
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
    >
      {"namedChunks":["en-translations"]}
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[0]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[1]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[2]"
    >
    </script>
    <script
      async
      data-chunk="en-translations"
      src="scripts[3]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`translations should render en-PSEUDO post-hydration 1`] = `
SCRIPTS: [
  "/runtime-c90bedc6c1e41d593d60.js",
  "/298-e7bbd14d05c894fe24a5.js",
  "/main-952ac1a7a163db10b764.js",
  "/en-translations-b17efc1aa5d571ed56f2.js",
]
CSS: []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[0]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[1]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[2]"
    >
    <link
      data-chunk="en-translations"
      rel="preload"
      as="script"
      href="scripts[3]"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Hello sku
        Company name
      </div>
    </div>
    <script
      id="__SKU_CLIENT_CONTEXT__"
      type="application/json"
    >
      {"language":"en"}
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
    >
      [938]
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
    >
      {"namedChunks":["en-translations"]}
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[0]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[1]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[2]"
    >
    </script>
    <script
      async
      data-chunk="en-translations"
      src="scripts[3]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: 
   <body>
     <div id="app">
       <div>
-        Hello sku
-        Company name
+        [Ḩẽẽẽƚƚööö šķǚǚǚ] [Çööm̂ƥăăกี้ýý กี้ăăm̂ẽẽ]
       </div>
     </div>
     <script

`;

exports[`translations should render fr 1`] = `
SCRIPTS: [
  "/runtime-c90bedc6c1e41d593d60.js",
  "/298-e7bbd14d05c894fe24a5.js",
  "/main-952ac1a7a163db10b764.js",
  "/fr-translations-8d39ba177a70a2aa4224.js",
]
CSS: []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[0]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[1]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[2]"
    >
    <link
      data-chunk="fr-translations"
      rel="preload"
      as="script"
      href="scripts[3]"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Bonjour sku
        บริษัท
      </div>
    </div>
    <script
      id="__SKU_CLIENT_CONTEXT__"
      type="application/json"
    >
      {"language":"fr"}
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
    >
      [590]
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
    >
      {"namedChunks":["fr-translations"]}
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[0]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[1]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[2]"
    >
    </script>
    <script
      async
      data-chunk="fr-translations"
      src="scripts[3]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`translations should support query parameters 1`] = `
SCRIPTS: [
  "/runtime-c90bedc6c1e41d593d60.js",
  "/298-e7bbd14d05c894fe24a5.js",
  "/main-952ac1a7a163db10b764.js",
  "/en-translations-b17efc1aa5d571ed56f2.js",
]
CSS: []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      My Awesome Project
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[0]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[1]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[2]"
    >
    <link
      data-chunk="en-translations"
      rel="preload"
      as="script"
      href="scripts[3]"
    >
  </head>
  <body>
    <div id="app">
      <div>
        Hello sku
        Company name
      </div>
    </div>
    <script
      id="__SKU_CLIENT_CONTEXT__"
      type="application/json"
    >
      {"language":"en"}
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
    >
      [938]
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
    >
      {"namedChunks":["en-translations"]}
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[0]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[1]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[2]"
    >
    </script>
    <script
      async
      data-chunk="en-translations"
      src="scripts[3]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;
